src/hooks/useKeyboardNavigation.js

```diff
@@ -27,11 +29,71 @@ const useKeyboardNavigation = ({
 
   useEffect(() => {
     return tinykeys(emojiListRef.current, {
-      ArrowRight: focusNextEmoji,
-      ArrowLeft: focusPrevEmoji,
+      ArrowRight: moveRight,
+      ArrowLeft: moveLeft,
+      ArrowUp: moveUp,
+      ArrowDown: moveDown,
     });
   }, []);
 
+  const moveRight = () => {
+    navigateGrid(RIGHT);
+  };
+  const moveLeft = () => {
+    navigateGrid(LEFT);
+  };
+  const moveUp = () => {
+    navigateGrid(UP);
+  };
+  const moveDown = () => {
+    navigateGrid(DOWN);
+  };
+
+  const updateActiveItem = next => {
+    next.firstChild.focus();
+  };
+
+  const navigateGrid = direction => {
+    const grid = getCurrentEmojiListGroup();
+    const active = getActiveElement().parentElement;
+    const gridChildren = Array.from(grid.children);
```

Comment:
This could be a function that returns a useMemo value, based on the filter result and the current grid. This way we don't need to Array.from on every keystroke.